app/controllers/participants/validations_controller.rb

```diff
@@ -0,0 +1,224 @@
+    before_action :set_form
+    before_action :check_not_already_completed, except: %i[complete reset]
+    before_action :validate_request_or_render, only: %i[do_you_know_your_trn
+                                                        have_you_changed_your_name
+                                                        confirm_updated_record
+                                                        name_not_updated
+                                                        tell_us_your_details
+                                                        confirm_details
+                                                        cannot_find_details]
+
+    def start
+      store_form_and_redirect_to_step :do_you_know_your_trn
+    end
+
+    def do_you_know_your_trn
+      choice = @participant_validation_form.do_you_know_your_trn_choice
+      if choice == "yes"
+        store_form_and_redirect_to_step :have_you_changed_your_name
+      elsif choice == "no"
+        store_form_and_redirect_to_step :find_your_trn
+      else
+        store_form_and_redirect_to_step :get_a_trn
+      end
+    end
+
+    def find_your_trn; end
+
+    def get_a_trn; end
+
+    def have_you_changed_your_name
+      choice = @participant_validation_form.have_you_changed_your_name_choice
+      if choice == "yes"
+        store_form_and_redirect_to_step :confirm_updated_record
+      else
+        store_form_and_redirect_to_step :tell_us_your_details
+      end
+    end
+
+    def confirm_updated_record
+      choice = @participant_validation_form.updated_record_choice
+      if choice == "yes"
+        store_form_and_redirect_to_step :tell_us_your_details
+      elsif choice == "no"
+        store_form_and_redirect_to_step :name_not_updated
+      else
+        store_form_and_redirect_to_step :check_with_tra
+      end
+    end
+
+    def name_not_updated
+      choice = @participant_validation_form.name_not_updated_choice
+      if choice == "register_previous_name"
+        store_form_and_redirect_to_step :tell_us_your_details
+      else
+        store_form_and_redirect_to_step :change_your_details_with_tra
+      end
+    end
+
+    def change_your_details_with_tra; end
+
+    def check_with_tra; end
+
+    def tell_us_your_details
+      store_form_and_redirect_to_step :confirm_details
+    end
+
+    def confirm_details
+      validate_participant_details_and_redirect
+    end
+
+    def cannot_find_details
+      store_validation_data!
+      reset_form_data
+      store_form_and_redirect_to_step :complete
+    end
+
+    def complete
+      @school = participant_profile.school
+      if participant_profile.ecf_participant_eligibility&.eligible_status?
+        # TRN has been validated, qts, no flags, not done before, no previous induction
+        render_completed_page
+      else
+        # all other cases
+        render_manual_check_page
+      end
+    end
+
+    def reset
+      reset_form_data
+      redirect_to participants_validation_start_path
+    end
```

Comment:
When do we use this?

Reply:
See above, just a manual way to reset your journey session variable. I thought it would help for testing the scenarios but we can remove it now or later?

Reply:
Let's at least disable it in production - probably worth removing all-together

Reply:
Removed it